worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -166,31 +171,43 @@ public long getSortedSize() {
     return sortedFilesSize.get();
   }
 
+  // For non-range requests:
+  //   1. If the sorted file is not generated, it returns the original unsorted FileInfo.
+  //   2. If the sorted file is generated, it returns the sorted FileInfo.
+  //
+  // For range requests:
+  //   1. If the sorted file is not generated, it adds the FileSorter task to the sorting queue and
+  //      synchronously waits for the sorted FileInfo.
+  //   2. If the FileSorter task is already in the sorting queue but the sorted file has not been
+  //      generated, it awaits until a timeout occurs (default 220 seconds).
+  //   3. If the sorted file is generated, it returns the sorted FileInfo.
   public FileInfo getSortedFileInfo(
       String shuffleKey, String fileName, FileInfo fileInfo, int startMapIndex, int endMapIndex)
       throws IOException {
     String fileId = shuffleKey + "-" + fileName;
     UserIdentifier userIdentifier = fileInfo.getUserIdentifier();
-
+    // not a range read and the sorted file not generate now
+    if (endMapIndex == Integer.MAX_VALUE && !sortedShuffleFiles.containsKey(shuffleKey)) {
```

Comment:
To check whether the sorted file has been generated, should also check the sorted set
```
    if (endMapIndex == Integer.MAX_VALUE && !sortedShuffleFiles.containsKey(shuffleKey) &&
      !sortedShuffleFiles.get(shuffleKey).contains(fileId)) {
      return fileInfo;
    }
```
Or else, do not call `getSortedFileInfo` if endMapIndex is max int.

Reply:
improved using Option 2.